### PR TITLE
Basic nix derivation

### PR DESCRIPTION
**Pull request by pfgray**
This commit includes a nix derivation to build tabry-rs and a devShell for easier development

### CODE REVIEW

File: default.nix
Comment (evanbattaglia) [inline default.nix]: Is that supposed to match the version in Cargo.toml? Not that at matters at this point, just wondering. When I bump versions I have to change both right? I'll add a comment about that in the Cargo.toml, or you can

File: default.nix
Comment (evanbattaglia) [inline default.nix]: The cargo package (and thus, resulting binary) is just called tabry, as it is intended to replace the old tabry executable (tabry ruby is just a library). Not sure what this should be called. I'm also not sure how I'm going to name the repos long term.

File: flake.nix
Comment (evanbattaglia) [inline flake.nix:21]: I also use clippy, cargo-llvm-cov, and bacon

File: flake.nix
Comment (evanbattaglia) [inline flake.nix]: TODO here?

File: default.nix
Comment (pfgray) [reply default.nix]: Oh, good question... I'm not sure, actually

File: default.nix
Comment (pfgray) [reply default.nix]: Ok, I'll change it to `tabry`, I don't think it matters too much unless we wanted to move it to nixpkgs. When I import it into my home-manager config, I import it as "`tabry`"

File: flake.nix
Comment (pfgray) [reply flake.nix:21]: Ah okay, are those baked into cargo commands? Does cargo handle pulling those, or do they need to be installed/maintained separately (for which they could be added to this list)? I'm not too familiar with the ecosystem yet

File: flake.nix
Comment (pfgray) [reply flake.nix]: Yeah, I can remove these... I just copied this from another project, and I leave this blank because I always end up forgetting the name of the attribute 😅

File: default.nix
Comment (pfgray) [reply default.nix]: It looks like it's included in the hash for the dependencies, so it makes sense to update it lock & step with `Cargo.toml`, I'll add a comment there

File: default.nix
Comment (evanbattaglia) [reply default.nix]: Have you ever gotten anything (either your own program or someone else's) in nixpkgs? I wonder what the process is like.
If I get a few more people using tabry, I may try :) Although higher priority would probably be brew since that is probably more common...

File: flake.nix
Comment (evanbattaglia) [reply flake.nix:21]: They are installed separately, although clippy and llvm-cov are run through cargo. When not using nix, clippy and cargo-llvm-cov are usually installed using rustup I believe, and bacon is installd using 'cargo install bacon', but there are probably nix derivations for all of them. Bacon is really nice, it watches your files and runs tests, builds, or runs clippy (linter)... you don't have to tel it which files or remember any command line arguments, you just type 'bacon'.

**Review by evanbattaglia — APPROVED** [review]
Thank you! I will merge once the fish commit it's based on is merged (github UI can't merge a later commit in a stack of commits like gerrit?!)

**Pull request merged**
